### PR TITLE
Jenkins version upgraded to 2.190

### PR DESCRIPTION
**Pull request by mhenschke-atl**
This is necessary for our SSH work, but I've been warned about version issues even on master, so we should update our latest version (and document it) before we release 1.1.1

### CODE REVIEW

**Comment by mhenschke-atl** [comment]
This issue includes the rather infamous java 11 mockito issue, that will need to be resolved before this can merge.

File: pom.xml [hunk 1]
```diff
@@ -165,6 +165,26 @@
...
+            <version>1.7.26</version>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>jcl-over-slf4j</artifactId>
+            <version>1.7.26</version>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>log4j-over-slf4j</artifactId>
```
Comment (dkjellin) [inline pom.xml:180]: Why is this here? We're not using log4j anywhere?

File: pom.xml [hunk 2]
```diff
@@ -165,6 +165,26 @@
...
+            <version>1.7.26</version>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
```
Comment (dkjellin) [inline pom.xml:184]: I'm confused why the sudden addition of `slf4j` dependencies, and none of them are from dependency management. 
Could you elaborate?

Comment (mhenschke-atl) [reply pom.xml:184]: Upper bound dependency issue. Looks like bumping the parent pom version solves it, so I'll do that instead (it creates the issue for hamcrest though)

File: pom.xml [hunk 3]
```diff
@@ -405,6 +419,15 @@
                 </plugins>
             </reporting>
         </profile>
+        <profile>
```
Comment (mhenschke-atl) [inline pom.xml:422]: So I've isolated the reason our unit tests are breaking in J11 down to the injected tests. I haven't gone over them thoroughly, but they appear mostly to be for things like Jelly formatting https://groups.google.com/g/jenkinsci-dev/c/POdTEQKq89s. They're a bit slower to run, and so I'm now imagining them running in a separate profile- which means for full tests coverage you have to maven twice. Not ideal.
Very open to feedback on this very very hacky solution.

Comment (mhenschke-atl) [reply pom.xml:422]: I think to find a true fix for this we'll need to reach out to the Jenkins community, so this is a stop-gap to unblock us from releasing.

File: pom.xml [hunk 4]
```diff
@@ -176,6 +177,18 @@
...
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-core</artifactId>
+            <version>2.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-library</artifactId>
+            <version>2.2</version>
```
Comment (dkjellin) [inline pom.xml:189]: I _assume_ these are versioned together? If so can we please extract the version out into a property, so we can bump it in one place rather than two when it becomes time to upgrade

File: acceptance-tests/pom.xml [hunk 5]
```diff
@@ -16,7 +16,7 @@
         <plugin-under-test.name>atlassian-bitbucket-server-integration</plugin-under-test.name>
         <jenkins.acceptance-test-harness.version>1.71</jenkins.acceptance-test-harness.version>
         <httpclient.version>4.5.10</httpclient.version>
-        <jenkins.version>2.176.4</jenkins.version>
+        <jenkins.version>2.190.1</jenkins.version>
```
Comment (goravjoshi) [inline acceptance-tests/pom.xml:19]: Can you do a quick check to see if our acceptance test library version is corresponding to this one?

Comment (mhenschke-atl) [reply acceptance-tests/pom.xml:19]: Thanks, good idea

Comment (mhenschke-atl) [reply acceptance-tests/pom.xml:19]: Checked this out, https://github.com/jenkinsci/acceptance-test-harness/blob/master/pom.xml looks like acceptance tests are using a much older version of Jenkins. Acceptance tests are green on bamboo so I think we're okay to go ahead with the version bump.
